R/scale-view.r

```diff
@@ -136,7 +136,7 @@ ViewScale <- ggproto("ViewScale", NULL,
     if (self$is_discrete()) {
       self$scale$map(x, self$limits)
     } else {
-      self$scale$map(x, self$continuous_range)
+      x
```

Comment:
can you explain this change to me, just because it is not obvious to me :-)

Reply:
I recall having good reasons for this, but they were a bit complicated.

* The current axis guide has seperate logic for mapping (breaks of) discrete scales, but not continuous scales: https://github.com/tidyverse/ggplot2/blob/1d1f79560c23241999529a92178a913769b8d28e/R/guides-axis.R#L85
  This makes sense, because the discrete mapping converts the breaks to integers, and: https://github.com/tidyverse/ggplot2/blob/1d1f79560c23241999529a92178a913769b8d28e/R/scale-continuous.R#L123-L125
* However, the non-position continuous scales, let's say `colour`, still need to use mapping on both continuous and discrete scales to get valid colours instead of numbers.
* So, if I use the axis logic, I get problems with not mapping continuous colour scales. If I use the legend/colourbar logic, I map the continuous position scales, which I needn't do (and was slowing down code somewhat).
* I was contorting the logic in the code, which I'm sure you'd agree is very unsatisfactory:
    * If my scale is discrete --> Use `map()`
    * If my scale is continuous:
        * If the aesthetic is *not* x/y --> Use `map()`
        * If the aesthetic *is* x/y --> Do not use `map()`
* Then, it occurred to me: I could just use `map()` for everything, if the continuous viewscales simply didn't do any mapping. They aren't used for anything else than building axes, so everything will be dandy.

Hence, this change was born and the `extract_key()` method now has nice and clean logic.

Reply:
sounds good :-)